### PR TITLE
Add edge cases to library

### PR DESCRIPTION
**Pull request by mariocao**
This PR adds some checks in order to avoid invalid inputs into the public functions.
Gas consumption increment is negligible.

### CODE REVIEW

File: contracts/Secp256k1.sol [hunk 1]
```diff
@@ -38,14 +38,15 @@ contract Secp256k1 is EllipticCurve {
     );
   }
 
-  /// @dev Function to derive the `y` coordinate given the `x` coordinate and the parity byte.
-  /// @param _yBit The parity byte following the ec point compressed format
+  /// @dev Function to derive the `y` coordinate given the `x` coordinate and the parity byte (`0x03` odd `y` and `0x04` even `y`).
```
Comment (aesedepece) [inline contracts/Secp256k1.sol]:
```suggestion
  /// @dev Function to derive the `y` coordinate given the `x` coordinate and the parity byte (`0x03` for odd `y` and `0x04` for even `y`).
```

File: test/vrf.js [hunk 2]
```diff
@@ -1,29 +1,41 @@
...
-        assert(coord[0].eq(web3.utils.toBN(point.uncompressed.x)))
-        assert(coord[1].eq(web3.utils.toBN(point.uncompressed.y)))
+        const coord = await vrf.decodePoint.call(web3.utils.hexToBytes(test.compressed))
+        assert(coord[0].eq(web3.utils.toBN(test.uncompressed.x)))
+        assert(coord[1].eq(web3.utils.toBN(test.uncompressed.y)))
       })
     }
-    for (let [index, test] of testdata.fastVerify.valid.entries()) {
+    for (let [index, test] of data.points.invalid.entries()) {
+      it(`should not decode a malformed compressed EC Point (${index + 1}) - ${test.description}`, async () => {
```
Comment (aesedepece) [inline test/vrf.js]: Negative tests can be better named by the source of the failure. E.g. `should fail to decode a compressed EC point if malformed`

File: test/vrf.js [hunk 3]
```diff
@@ -39,24 +51,43 @@ contract("VRF", accounts => {
         assert(params[1][3].eq(web3.utils.toBN(test.vComponents.cGamma.y)))
       })
     }
+    for (let [index, test] of data.computeFastVerifyParams.invalid.entries()) {
+      it(`should not compute fast verify parameters (${index + 1}) - ${test.description}`, async () => {
```
Comment (aesedepece) [inline test/vrf.js]: This test could use better naming too.

File: test/vrf.js [hunk 4]
```diff
@@ -39,24 +51,43 @@ contract("VRF", accounts => {
...
+        const publicKey = await vrf.decodePoint.call(web3.utils.hexToBytes(test.pub))
         const proof = await vrf.decodeProof.call(web3.utils.hexToBytes(test.pi))
         const message = web3.utils.hexToBytes(test.message)
         const result = await vrf.verify.call(publicKey, proof, message)
         assert.equal(result, true)
       })
     }
-    for (let [index, test] of testdata.verify.invalid.entries()) {
+    for (let [index, test] of data.verify.invalid.entries()) {
       it(`should not verify a VRF proof (${index + 1}) - ${test.description}`, async () => {
```
Comment (aesedepece) [inline test/vrf.js]: This too.

File: test/vrf.js [hunk 5]
```diff
@@ -97,5 +128,35 @@ contract("VRF", accounts => {
         assert.equal(res, true)
       })
     }
+    for (let [index, test] of data.fastVerify.invalid.entries()) {
+      it(`should not fast verify an invalid VRF proof (${index + 1}) - ${test.description}`, async () => {
```
Comment (aesedepece) [inline test/vrf.js]: This too.

**Comment by mariocao** [comment]
All PR requested changes already included! :smile:

**Pull request merged**